### PR TITLE
Enabled options object to set domain and pathless-ness of a cookie

### PR DESCRIPTION
**Pull request by joeyjojo**
This is our initial pass at this work.

You may notice there are no tests for this... we struggled to think of tests that were worthwhile and not just testing the cookie implementation of the browser, which seemed pointless.

Happy to take a steer on this if you have a good idea.

Solves #284

### CODE REVIEW

**Comment by dhurrell** [comment]
You could create a stub setter on document.cookie (using Object.defineProperty) to ensure that the cookie string is generated correctly based on the domain and isPathless optional properties - but if you feel like that's overly convoluted and not worthwhile, I won't disagree :) I'm also not sure how you'd get the original document.cookie back...

Otherwise (aside from my other comment about JsDoc) it looks fine to me.

File: static/script/devices/storage/cookie.js [hunk 1]
```diff
@@ -25,107 +25,114 @@
...
+            date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
+            var cookieDataArray = [
+                    namespace + "=" + value,
+                    "expires=" + date.toGMTString(),
+                    "path=" + (opts.isPathless ? '/' : path)
+            ];
+            if (opts.domain){
+                cookieDataArray.push("domain="+opts.domain);
+            }
+            document.cookie = cookieDataArray.join('; ')
```
Comment (dhurrell) [inline static/script/devices/storage/cookie.js]: Semicolon me

**Comment by joeyjojo** [comment]
@dhurrell can you take another look please, we weren't entirely sure on the documentation style so hopefully its up to scratch.

**Comment by dhurrell** [comment]
Looks good to me!

**Comment by joeyjojo** [comment]
@dhurrell @sdclibbery can we get this merged please?

File: static/script/devices/storage/cookie.js [hunk 2]
```diff
@@ -25,107 +25,128 @@
...
-	}
-);
+                // delete it from the stored namespaces
+                // so it will be reloaded the next time
+                // we get it
+                delete namespaces[this._namespace];
+            },
+            _isEmpty: function() {
+                var prop;
+                for(prop in this._valueCache) {
```
Comment (sdclibbery) [inline static/script/devices/storage/cookie.js]: This for...in without any hasOwnProperty makes me uncomfortable, but it was in the original code and I don't propose we change it now.

**Pull request merged**